src/QMCWaveFunctions/TrialWaveFunction.h

```diff
@@ -480,7 +480,10 @@ class TrialWaveFunction
   ///diff of the phase of the trial wave function during ratio calls
   RealType PhaseDiff;
 
-  ///log of the trial wave function
+  /** log of the trial wave function
+   *  Naming covention of WFC and all its children is the LogValue type
+   *  is complex and contains PhaseValue as well. :(
+   */
   RealType LogValue;
```

Comment:
In keeping the scope of the PR limited I chose not to deal with the situation in TrialWaveFunction but only WaveFunctionComponent and down.  A future PR will address it. So far the type mess in TrialWaveFunction has not obstructed my work. 

This effort was brought on by conflicts about the type of CurRatio and LogValue's in WFC and downward in mixed precision. Which is also why I'm not interested in giving ground on `WaveFunctionTypes`.

Reply:
Right now the renaming, a non-functional change, just complicated this PR.
```
yeluo@epyc-server:~/opt/qmcpack/src$ grep -R LogValue |wc
    425    2188   43063
yeluo@epyc-server:~/opt/qmcpack/src$ grep -R LogValue QMCWaveFunctions/ |wc
    409    2077   41391
```

Reply:
Where it is a reference to a LogValue log_value_ it got renamed where it was a reference to a TWF.LogValue it didn't.  I did this using the AST from clang and then checked it by hand with wgrep.

TWF.LogValue and WFC.LogValue were never the same type and that is no worse than it was before.

Reply:
> TWF.LogValue and WFC.LogValue were never the same type and that is no worse than it was before.

This is true. My concern is about renaming. Prefer not leaving any LogValue as a type. Do the change in one shot.

Reply:
I'll look at it this afternoon, That's above the dividing line where I need to fix the configuration time type locking so I haven't dug into it yet.  I will try and just rename the variable which shouldn't cause mayhem.  I'd like to leave the composed type via WaveFunctionTypes and the inherited definition with distinct names. For now the locked types almost universally have the redundant `Type` or `_t` which I'd keep on them.

Reply:
I corrected my previous comment Prefer not leaving any LogValue as a ~~type~~ value since LogValue is a type based on our naming rule. Sorry for the confusion.